Implement example app using hydra-sdk

vrom911 opened the pull request:
Core Features

* WebSocket connection to hydra-node
* Real-time connection status indicator
* Automatic history replay on connect
* Generate new wallet - Creates Ed25519 keypair in-browser
* Restore from private key - Restore using 64-char hex key
* Load from .sk file - Import cardano-cli key files
* Export keys - Download cardano.sk and cardano.vk files
* L1 balance display with UTxO breakdown
* UTxO splitting - Split large UTxOs into smaller ones for better commit flexibility
* Generate Hydra signing/verification keys using SDK's `KeysUtils.hydraCliKeygen()`
* Export hydra.sk and hydra.vk files for hydra-node configuration
* Modal with selectable L1 UTxOs
* Auto-filters script-locked UTxOs and native assets
* Commits
* Send tx between parties inside the Head
* Real-time head balance display
* Aborting
* Closing
* Fanout

https://github.com/user-attachments/assets/02422855-6906-4755-8177-03d5427b95b6


https://github.com/user-attachments/assets/9b181ae8-e169-486c-859f-6e3154127d7a




<!-- Describe your change here -->

---

<!-- Consider each and tick it off one way or the other -->
* [ ] CHANGELOG updated or not needed
* [ ] Documentation updated or not needed
* [ ] Haddocks updated or not needed
* [ ] No new TODOs introduced or explained herafter

vrom911 committed:
Implement example app using hydra-sdk

aniadev commented:
@vrom911 You can use hydra-sdk to generate `hydra.sk` and `hydra.vk`.
The SDK supports this method starting from version v1.1.5:
https://hydrasdk.com/api/utilities#keysutilshydraclikeygen-hydraclikeypair

vrom911 committed:
Generate also hydra keys

vrom911 committed:
Fixes of tx

noonio committed:
Nix flake; minor fixup

noonio commented:
This is great!

As discussed, we're going to put this in a new repo.

noonio committed:
Nix flake; minor fixup (#2415)

- Add a nix flake (because this will be in a new repo)
- Fix small calculation around maximum amount to send

vrom911 commented:
It lives here now: https://github.com/cardano-scaling/hydra-sdk-app
